### DISCUSSION TITLE
🚀 Distro 2020.05.patch-01 Released

### DISCUSSION THREAD

**Discussion by Nicogene** [robotology/community#425] [Releases]
Hi @robotology/everyone 👋 

We're happy to announce that our latest iCub software distro **2020.05.patch-01** is now available.

To download the source code with the correct tags, check out our up-to-date [Software Versioning Table](https://wiki.icub.org/wiki/Software_Versioning_Table).

⚠️ This distro impacts only on `icub-firmware` and `icub-firmware-build` repositories.

This patch distro has been released after we spotted (and then fixed) a bug in the firmware of the bootloader of some CAN boards (`strain2`, `mtb4`, `rfe`, `psc`) released in the distro `2020.05` (#404).

This bug affects the bootloader making impossible a firmware update of the applications of those board. 

Like the feature-based releases, patches are shipped only in terms of tags/branches.

**Comment by DanielePucci** [comment]
CC @robotology/iit-dynamic-interaction-control

**Comment by traversaro** [comment]
> This patch distro has been released after we spotted (and then fixed) a bug in the firmware of the bootloader of some CAN boards (`strain2`, `mtb4`, `rfe`, `psc`) released in the distro `2020.05` (#404).

Hi @Nicogene, how  is this bug affecting the behaviour of the boards? Users that are using the `2020.05` distro version/`1.16.0` `icub-firmware` version with no problem till now are suggested to upgrade (because the problem could affected them) or they can continue to use the `2020.05`/`1.16.0` firmware?

**Comment by Nicogene** [comment]
@marcoaccame said:

> This bug does not affect the behavior of the application because the CAN parser of the application does not use the frame length but only its content.

> However, the bug does affect the behavior of the bootloader and of the `updaterofbootloader` because a message used for firmware update over CAN does use the length.

> Hence, if a board `strain2`, `mtb4`, `rfe` or `psc` have a bootloader released with 2020.05 then it is not possible to perform firmware updates on them.

This means that the applications of the firmware `v1.16.0` are fine but not upgradable unless upgrading first the `*bootloader.hex`s to `v1.16.1`.
 They can continue to use that firmware unless they are not planning to upgrade it to the `v1.17.0` that will be released soon.
(Please @marcoaccame correct me if I am wrong)
